application_sandbox/conditional_rendering/main.cpp

```diff
@@ -0,0 +1,527 @@
+
+    cmdBuffer->vkCmdBeginConditionalRenderingEXT(cmdBuffer,
+                                                 &conditional_begin3);
+    cmdBuffer->vkCmdBindPipeline(cmdBuffer, VK_PIPELINE_BIND_POINT_COMPUTE,
+                                 *compute_pipeline_);
+    cmdBuffer->vkCmdBindDescriptorSets(
+        cmdBuffer, VK_PIPELINE_BIND_POINT_COMPUTE,
+        ::VkPipelineLayout(*compute_pipeline_layout_), 0, 1,
+        &frame_data->compute_descriptor_set_->raw_set(), 0, nullptr);
+    cmdBuffer->vkCmdDispatch(cmdBuffer, 1, 1, 1);
+    cmdBuffer->vkCmdEndConditionalRenderingEXT(cmdBuffer);
+
+    cmdBuffer->vkCmdPipelineBarrier(cmdBuffer,
+                                    VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT,
+                                    VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT, 0, 0,
+                                    nullptr, 1, &to_use_in_frag, 0, nullptr);
+
+    cmdBuffer->vkCmdBeginRenderPass(cmdBuffer, &pass_begin,
+                                    VK_SUBPASS_CONTENTS_INLINE);
+
+    cmdBuffer->vkCmdBindPipeline(cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
+                                 *cube_pipeline_);
+    cmdBuffer->vkCmdBindDescriptorSets(
+        cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
+        ::VkPipelineLayout(*pipeline_layout_), 0, 1,
+        &frame_data->cube_descriptor_set_->raw_set(), 0, nullptr);
+
+    VkClearColorValue clear_color{0.0f, 1.0f, 1.0f, 1.0f};
+    VkClearAttachment attachment = {
+        VK_IMAGE_ASPECT_COLOR_BIT,
+        0,
+        clear_color,
+    };
+    VkClearRect rect = {
+        {
+            {0, 0},
+            {app()->swapchain().width(), app()->swapchain().height()},
+        },
+        0,  // baseArrayLayer
+        1,  // layerCount
+    };
+
+    cmdBuffer->vkCmdBeginConditionalRenderingEXT(cmdBuffer,
+                                                 &conditional_begin1);
+    cmdBuffer->vkCmdClearAttachments(cmdBuffer, 1, &attachment, 1, &rect);
+    cmdBuffer->vkCmdEndConditionalRenderingEXT(cmdBuffer);
+
+    cmdBuffer->vkCmdBeginConditionalRenderingEXT(cmdBuffer,
+                                                 &conditional_begin2);
+    VkClearColorValue clear_color2{1.0f, 0.0f, 1.0f, 1.0f};
+    attachment.clearValue.color = clear_color2;
+    cmdBuffer->vkCmdClearAttachments(cmdBuffer, 1, &attachment, 1, &rect);
+    cube_.Draw(&cmdBuffer);
+    cmdBuffer->vkCmdEndConditionalRenderingEXT(cmdBuffer);
+
+    cmdBuffer->vkCmdEndRenderPass(cmdBuffer);
+
+    (*frame_data->command_buffer_)
+        ->vkEndCommandBuffer(*frame_data->command_buffer_);
+  }
+
+  virtual void Update(float time_since_last_render) override {
+    frames_since_last_notify_ += 1;
+    model_data_->data().transform =
+        model_data_->data().transform *
+        Mat44::FromRotationMatrix(
+            Mat44::RotationX(3.14f * time_since_last_render) *
+            Mat44::RotationY(3.14f * time_since_last_render * 0.5f));
+
+    conditional_data_->data().first_condition =
```

Comment:
When testing an extension I'd suggest having the first frames draw something rather than be blank and also show off the extension.  So maybe show one cube, then the other, then both?

Reply:
Got rid of the second condition, just one now. First will show blue background with 2 cubes (2 instances) and then a single cube